frontend/src/components/teacher/session-creation/DistributeAssessmentButton.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { useMutation } from "@apollo/client";
+import { Button } from "@chakra-ui/react";
+
+import { CREATE_TEST_SESSION } from "../../../APIClients/mutations/TestSessionMutations";
+import { GET_TEST_SESSIONS_BY_TEACHER_ID } from "../../../APIClients/queries/TestSessionQueries";
+import type { TestSessionRequest } from "../../../APIClients/types/TestSessionClientTypes";
+import { PaperPlaneOutlineIcon } from "../../../assets/icons";
+import { DISPLAY_ASSESSMENTS_PAGE } from "../../../constants/Routes";
+import Toast from "../../common/info/Toast";
+
+interface DistributeAssessmentButtonProps {
+  testSessionRequest: TestSessionRequest;
+}
+
+const DistributeAssessmentButton = ({
+  testSessionRequest,
+}: DistributeAssessmentButtonProps): React.ReactElement => {
+  const history = useHistory();
+  const { showToast } = Toast();
```

Comment:
this is the nittiest of nits and not related to your pr, but any chance we could rename `Toast` to `useToast`? It's a custom hook but we're not using a custom-hook name name for it and that's been driving me crazy 😆 (for a more concrete reason why it might be nice, this gets us linter rules which check for rules-of-hooks, which can help catch bugs)

Reply:
lol sure